corehq/ex-submodules/pillowtop/reindexer/change_providers/case.py

```diff
@@ -0,0 +1,49 @@
+import itertools
+
+from casexml.apps.case.models import CommCareCase
+from corehq.form_processor.backends.sql.dbaccessors import CaseAccessorSQL
+from corehq.form_processor.change_providers import _sql_case_to_change
+from corehq.form_processor.utils.general import should_use_sql_backend
+from pillowtop.reindexer.change_providers.couch import CouchViewChangeProvider
+from pillowtop.reindexer.change_providers.interface import ChangeProvider
+
+
+class CouchDomainCaseChangeProvider(CouchViewChangeProvider):
+    def __init__(self, domain):
+        self.document_class = CommCareCase
+        self._couch_db = self.document_class.get_db()
+        self._view_name = 'cases_by_owner/view'
+        self._chunk_size = 100
+        self._view_kwargs = {
+            'include_docs': True,
+            'startkey': [domain],
+            'endkey': [domain, {}, {}]
+        }
+
+
+class SqlDomainCaseChangeProvider(ChangeProvider):
+
+    def __init__(self, domain):
+        self.domain = domain
+
+    def iter_changes(self, start_from=None):
+        case_ids = CaseAccessorSQL.get_case_ids_in_domain(self.domain)
+        for case in CaseAccessorSQL.get_cases(case_ids):
+            yield _sql_case_to_change(case)
+
+
+class DomainCaseChangeProvider(ChangeProvider):
```

Comment:
non-blocking, but I think this could be generalized to a change provider of change providers and then the details of constructing this particular instance of that could be made a function